pkg/apiserver/registry/inventory/virtualmachine/rest.go

```diff
@@ -197,3 +180,148 @@ func (r *REST) Watch(ctx context.Context, options *internalversion.ListOptions)
 	key, label, field := store.GetSelectors(options)
 	return r.cloudInventory.WatchVms(ctx, key, label, field)
 }
+
+// sortAndConvertObjsToVmList sorts the objs based on Namespace and Name and returns the VPC list.
+func sortAndConvertObjsToVmList(objs []interface{}) *runtimev1alpha1.VirtualMachineList {
+	sort.Slice(objs, func(i, j int) bool {
+		vmI := objs[i].(*runtimev1alpha1.VirtualMachine)
+		vmJ := objs[j].(*runtimev1alpha1.VirtualMachine)
+		// First Sort with Namespace.
+		if vmI.Namespace < vmJ.Namespace {
+			return true
+		} else if vmI.Namespace > vmJ.Namespace {
+			return false
+		}
+		// Second Sort with Name.
+		return vmI.Name < vmJ.Name
+	})
+
+	vmList := &runtimev1alpha1.VirtualMachineList{}
+	for _, obj := range objs {
+		vm := obj.(*runtimev1alpha1.VirtualMachine)
+		vmList.Items = append(vmList.Items, *vm)
+	}
+	return vmList
+}
+
+// getSelectors creates and returns a map of supported label and field selectors.
+func getSelectors(options *internalversion.ListOptions) (map[string]string, map[string]string, error) {
+	var validLabelSelectors, validFieldSelectors map[string]string
+
+	validLabelSelectors = make(map[string]string, numLabelSelectors)
+	validFieldSelectors = make(map[string]string, numFieldSelectors)
+	if err := getLabelSelectors(options, validLabelSelectors); err != nil {
+		return nil, nil, err
```

Comment:
Are these error user friendly?

Reply:
Yes.. The error will be something. Same of Field selectors as well.
unsupported label selector, supported labels are: "+
					"%v and %v", labels.CloudAccountName, labels.CloudAccountNamespace